app/models/recipients.rb

```diff
@@ -17,18 +17,23 @@ def user_emails
 
   def users
     return @users if @users
-    @users = User.joins(:teams, :roles).where(roles: { name: roles })
-    if group.to_sym == :selected_teams
-      @users = @users.where(teams: { kind: Team::KINDS })
-    elsif group.to_sym == :unselected_teams
-      @users = @users.where(teams: { kind: nil })
-    end
-    if seasons.any?
-      @users = @users.where(teams: {
-        season: Season.where(name: seasons).pluck(:id)
-      })
-    end
-    @users.uniq
+    @users = User.joins(:roles).where(roles: { name: roles })
+    @users = @users.where('roles.team_id IN (?) OR roles.name IN (?)', teams, teamless_roles) if filter_by_teams?
+    @users = @users.uniq
+    @users
+  end
+
+  def filter_by_teams?
+    group.to_sym != :everyone || seasons.any?
+  end
+
+  def teams
+    return @teams if @teams
+    @teams = Team
+    @teams = @teams.where(kind: group.to_sym == :selected_teams ? Team::KINDS : nil) if group.to_sym != :everyone
```

Comment:
`kind: nil` selects every user in the system that isn't part of a team that hasn't been accepted as either sponsored or voluntary team ever, including all applicants whose applications had been rejected.  Is that the intended behaviour?